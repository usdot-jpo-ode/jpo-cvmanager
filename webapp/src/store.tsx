import { Action, ThunkAction, configureStore } from '@reduxjs/toolkit'
import rsuReducer from './generalSlices/rsuSlice'
import userReducer from './generalSlices/userSlice'
import wzdxReducer from './generalSlices/wzdxSlice'
import configReducer from './generalSlices/configSlice'
import intersectionReducer from './generalSlices/intersectionSlice'
import adminAddOrganizationReducer from './features/adminAddOrganization/adminAddOrganizationSlice'
import adminAddRsuReducer from './features/adminAddRsu/adminAddRsuSlice'
import adminAddIntersectionReducer from './features/adminAddIntersection/adminAddIntersectionSlice'
import adminAddUserReducer from './features/adminAddUser/adminAddUserSlice'
import adminEditOrganizationReducer from './features/adminEditOrganization/adminEditOrganizationSlice'
import adminEditRsuReducer from './features/adminEditRsu/adminEditRsuSlice'
import adminEditIntersectionReducer from './features/adminEditIntersection/adminEditIntersectionSlice'
import adminEditUserReducer from './features/adminEditUser/adminEditUserSlice'
import adminOrganizationTabReducer from './features/adminOrganizationTab/adminOrganizationTabSlice'
import adminOrganizationTabUserReducer from './features/adminOrganizationTabUser/adminOrganizationTabUserSlice'
import adminOrganizationTabRsuReducer from './features/adminOrganizationTabRsu/adminOrganizationTabRsuSlice'
import adminOrganizationTabIntersectionReducer from './features/adminOrganizationTabIntersection/adminOrganizationTabIntersectionSlice'
import adminRsuTabReducer from './features/adminRsuTab/adminRsuTabSlice'
import adminIntersectionTabReducer from './features/adminIntersectionTab/adminIntersectionTabSlice'
import adminUserTabReducer from './features/adminUserTab/adminUserTabSlice'
import adminNotificationTabReducer from './features/adminNotificationTab/adminNotificationTabSlice'
import adminAddNotificationReducer from './features/adminAddNotification/adminAddNotificationSlice'
import adminEditNotificationReducer from './features/adminEditNotification/adminEditNotificationSlice'
import menuReducer from './features/menu/menuSlice'
import asn1DecoderSlice from './features/intersections/decoder/asn1-decoder-slice'
import intersectionMapReducer from './features/intersections/map/map-slice'
import intersectionMapLayerStyleReducer from './features/intersections/map/map-layer-style-slice'
import dataSelectorReducer from './features/intersections/data-selector/dataSelectorSlice'
import mapSliceReducer from './pages/mapSlice'

export const setupStore = (preloadedState: any) => {
  return configureStore({
    reducer: {
      rsu: rsuReducer,
      user: userReducer,
      wzdx: wzdxReducer,
      config: configReducer,
      intersection: intersectionReducer,
      adminAddOrganization: adminAddOrganizationReducer,
      adminAddRsu: adminAddRsuReducer,
      adminAddIntersection: adminAddIntersectionReducer,
      adminAddUser: adminAddUserReducer,
      adminEditOrganization: adminEditOrganizationReducer,
      adminEditRsu: adminEditRsuReducer,
      adminEditIntersection: adminEditIntersectionReducer,
      adminEditUser: adminEditUserReducer,
      adminOrganizationTab: adminOrganizationTabReducer,
      adminOrganizationTabUser: adminOrganizationTabUserReducer,
      adminOrganizationTabRsu: adminOrganizationTabRsuReducer,
      adminOrganizationTabIntersection: adminOrganizationTabIntersectionReducer,
      adminRsuTab: adminRsuTabReducer,
      adminIntersectionTab: adminIntersectionTabReducer,
      adminUserTab: adminUserTabReducer,
      adminNotificationTab: adminNotificationTabReducer,
      adminAddNotification: adminAddNotificationReducer,
      adminEditNotification: adminEditNotificationReducer,
      menu: menuReducer,
      intersectionMap: intersectionMapReducer,
      intersectionMapLayerStyle: intersectionMapLayerStyleReducer,
      dataSelector: dataSelectorReducer,
      map: mapSliceReducer,
      asn1Decoder: asn1DecoderSlice,
    },
    preloadedState,
    middleware: (getDefaultMiddleware) =>
      getDefaultMiddleware({
        thunk: true,
        serializableCheck: false,
        immutableCheck: false,
      }),
    devTools: true,
  })
}

type AppStore = ReturnType<typeof setupStore>
export type AppState = ReturnType<AppStore['getState']>
type AppThunk<ReturnType = void> = ThunkAction<ReturnType, AppState, unknown, Action>

export type RootState = ReturnType<ReturnType<typeof setupStore>['getState']>
