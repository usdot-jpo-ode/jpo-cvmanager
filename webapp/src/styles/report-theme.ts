import { createTheme } from '@mui/material'

export const ReportTheme = createTheme({
  typography: {
    h1: {
      color: 'black',
    },
    h2: {
      color: 'black',
    },
    h3: {
      color: 'black',
    },
    h4: {
      color: 'black',
    },
    h5: {
      color: 'black',
    },
    h6: {
      color: 'black',
    },
    body1: {
      color: 'black',
    },
    body2: {
      color: 'black',
    },
  },
  palette: {
    mode: 'light',
    primary: {
      main: '#000000',
      light: '#000000',
      dark: '#000000',
      contrastText: '#FFFFFF',
    },
    secondary: {
      main: '#000000',
      light: '#000000',
      dark: '#000000',
      contrastText: '#FFFFFF',
    },
    error: {
      main: '#ad2626',
    },
    success: {
      main: '#2e942e',
    },
    text: {
      primary: '#030303',
      secondary: '#030303',
      disabled: 'rgba(66, 66, 66, 0.48)',
    },
    divider: '#575757',
    background: {
      default: '#ffffff',
      paper: '#FFFFFF',
    },
    custom: {
      mapLegendBackground: '#ffffff',
      tableHeaderBackground: '#ffffff',
      tableErrorBackground: '#ffffff',
      mapStyleFilePath: 'mapbox-styles/main-light.json',
      mapStyleHasTraffic: true,
      mapMenuBackground: '#ffffff',
      mapMenuItemBackgroundSelected: '#ffffff',
      mapMenuItemBorderSelected: '1px solid black',
      mapMenuItemHoverSelected: '#a19f9f',
      mapMenuItemHoverUnselected: '#ffffff',
      rowActionIcon: '#9DBDD3',
    },
  },
})
