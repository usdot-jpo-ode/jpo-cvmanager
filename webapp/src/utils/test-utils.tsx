/**
 * This function searches for the every somewhat random/chaotic class name and property that cause snapshot tests to be inconsistent.
 * This current list includes MUI classes (css-*) and aria-invalid attributes.
 *
 * @ex :
 * ```
 * const { container } = render(<Component />);
 *
 * replaceChaoticIds(container);
 * ```
 *
 * @param container The HTMLElement node to search for SSR ids
 */
function replaceChaoticIds(container: HTMLElement) {
  const props = [
    {
      selector: 'class',
      updateFunc: (val: string) => val.replace(/css-[0-9a-z]*?-/g, 'css-mocked-'),
    },
  ]

  container.querySelectorAll('*').forEach((item) => {
    props.forEach((prop) => {
      if (item.getAttribute(prop.selector)) {
        item.setAttribute(prop.selector, prop.updateFunc(item.getAttribute(prop.selector)))
      }
    })
  })
  container.querySelectorAll('input[aria-invalid]').forEach((item) => {
    item.removeAttribute('aria-invalid')
  })
  return container
}

import React from 'react'
import { LocalizationProvider } from '@mui/x-date-pickers/LocalizationProvider'
import { AdapterDayjs } from '@mui/x-date-pickers/AdapterDayjs'
import dayjs from 'dayjs'
import utc from 'dayjs/plugin/utc'
import timezone from 'dayjs/plugin/timezone'

// Extend dayjs with timezone and UTC plugins
dayjs.extend(utc)
dayjs.extend(timezone)

// Mock component to set timezone
const MockLocalizationProvider = ({ children }) => {
  dayjs.tz.setDefault('America/Denver')
  return <LocalizationProvider dateAdapter={AdapterDayjs}>{children}</LocalizationProvider>
}

export { replaceChaoticIds, MockLocalizationProvider }
