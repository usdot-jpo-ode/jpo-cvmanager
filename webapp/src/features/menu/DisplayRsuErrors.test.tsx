import React from 'react'
import { render } from '@testing-library/react'
import DisplayRsuErrors from './DisplayRsuErrors'
import { Provider } from 'react-redux'
import { ThemeProvider } from '@mui/material'
import { testTheme } from '../../styles'
import { setupStore } from '../../store'
import { replaceChaoticIds } from '../../utils/test-utils'

jest.useFakeTimers().setSystemTime(new Date('2024-10-01'))

it('should take a snapshot', () => {
  const { container } = render(
    <ThemeProvider theme={testTheme}>
      <Provider store={setupStore({})}>
        <DisplayRsuErrors
          initialSelectedRsu={{
            id: 1,
            type: 'Feature',
            geometry: {
              type: 'Point',
              coordinates: [],
            },
            properties: {
              rsu_id: 1,
              milepost: 1,
              geography: 'POINT (39.7392 -104.9903)',
              model_name: 'model_name',
              ipv4_address: '1.1.1.1',
              primary_route: 'primary_route',
              serial_number: 'serial_number',
              manufacturer_name: 'manufacturer_name',
            },
          }}
        />
      </Provider>
    </ThemeProvider>
  )

  expect(replaceChaoticIds(container)).toMatchSnapshot()
})
