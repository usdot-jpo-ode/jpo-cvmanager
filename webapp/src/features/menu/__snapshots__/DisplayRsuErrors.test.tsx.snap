// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take a snapshot 1`] = `
<div>
  <div
    style="display: flex; justify-content: center; margin: auto; width: 100%; height: 100%; background-color: rgb(0, 0, 0); border-radius: 4px;"
  >
    <div
      class="MuiStack-root css-mocked-MuiStack-root"
    >
      <div
        class="MuiStack-root css-mocked-MuiStack-root"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 css-mocked-MuiTypography-root"
        >
          1.1.1.1 Status
        </p>
        <button
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-mocked-MuiButtonBase-root-MuiIconButton-root"
          tabindex="0"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-mocked-MuiSvgIcon-root"
            data-testid="CloseIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
            />
          </svg>
        </button>
      </div>
      <div
        class="accordion"
      >
        <div
          class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 MuiAccordion-root MuiAccordion-rounded MuiAccordion-gutters css-mocked-MuiPaper-root-MuiAccordion-root"
          style="--Paper-shadow: none;"
        >
          <h3
            class="MuiAccordion-heading css-mocked-MuiAccordion-heading"
          >
            <div
              aria-expanded="false"
              class="MuiButtonBase-root MuiAccordionSummary-root MuiAccordionSummary-gutters css-mocked-MuiButtonBase-root-MuiAccordionSummary-root"
              role="button"
              tabindex="0"
            >
              <div
                class="MuiAccordionSummary-content MuiAccordionSummary-contentGutters css-mocked-MuiAccordionSummary-content"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 css-mocked-MuiTypography-root"
                >
                  Online Status
                </p>
              </div>
              <div
                class="MuiAccordionSummary-expandIconWrapper css-mocked-MuiAccordionSummary-expandIconWrapper"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-mocked-MuiSvgIcon-root"
                  data-testid="ExpandMoreIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M16.59 8.59 12 13.17 7.41 8.59 6 10l6 6 6-6z"
                  />
                </svg>
              </div>
            </div>
          </h3>
          <div
            class="MuiCollapse-root MuiCollapse-vertical MuiCollapse-hidden css-mocked-MuiCollapse-root"
            style="min-height: 0px;"
          >
            <div
              class="MuiCollapse-wrapper MuiCollapse-vertical css-mocked-MuiCollapse-wrapper"
            >
              <div
                class="MuiCollapse-wrapperInner MuiCollapse-vertical css-mocked-MuiCollapse-wrapperInner"
              >
                <div
                  class="MuiAccordion-region"
                  role="region"
                >
                  <div
                    class="MuiAccordionDetails-root css-mocked-MuiAccordionDetails-root"
                  >
                    <div
                      style="background-color: rgb(0, 0, 0); border-top: 1px solid white; border-bottom: 1px solid white; font-family: \\"museo-slab\\" Arial Helvetica Sans-Serif; width: 90%; padding: 0.5rem 1rem;"
                    >
                      <p>
                        <b>
                          RSU Online Status: 
                        </b>
                        Offline
                      </p>
                      <br />
                      <p>
                        <b>
                          RSU Last Online: 
                        </b>
                        No Data
                      </p>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 MuiAccordion-root MuiAccordion-rounded MuiAccordion-gutters css-mocked-MuiPaper-root-MuiAccordion-root"
          style="--Paper-shadow: none;"
        >
          <h3
            class="MuiAccordion-heading css-mocked-MuiAccordion-heading"
          >
            <div
              aria-expanded="false"
              class="MuiButtonBase-root MuiAccordionSummary-root MuiAccordionSummary-gutters css-mocked-MuiButtonBase-root-MuiAccordionSummary-root"
              role="button"
              tabindex="0"
            >
              <div
                class="MuiAccordionSummary-content MuiAccordionSummary-contentGutters css-mocked-MuiAccordionSummary-content"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 css-mocked-MuiTypography-root"
                >
                  SCMS Status
                </p>
              </div>
              <div
                class="MuiAccordionSummary-expandIconWrapper css-mocked-MuiAccordionSummary-expandIconWrapper"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-mocked-MuiSvgIcon-root"
                  data-testid="ExpandMoreIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M16.59 8.59 12 13.17 7.41 8.59 6 10l6 6 6-6z"
                  />
                </svg>
              </div>
            </div>
          </h3>
          <div
            class="MuiCollapse-root MuiCollapse-vertical MuiCollapse-hidden css-mocked-MuiCollapse-root"
            style="min-height: 0px;"
          >
            <div
              class="MuiCollapse-wrapper MuiCollapse-vertical css-mocked-MuiCollapse-wrapper"
            >
              <div
                class="MuiCollapse-wrapperInner MuiCollapse-vertical css-mocked-MuiCollapse-wrapperInner"
              >
                <div
                  class="MuiAccordion-region"
                  role="region"
                >
                  <div
                    class="MuiAccordionDetails-root css-mocked-MuiAccordionDetails-root"
                  >
                    <div
                      style="background-color: rgb(0, 0, 0); border-top: 1px solid white; border-bottom: 1px solid white; font-family: \\"museo-slab\\" Arial Helvetica Sans-Serif; width: 90%; padding: 0.5rem 1rem;"
                    >
                      <p>
                        <b>
                          SCMS Status: 
                        </b>
                        Unhealthy
                      </p>
                      <br />
                      <p>
                        <b>
                          SCMS Expiration: 
                        </b>
                        Never downloaded certificates
                      </p>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        style="display: flex; align-content: center; justify-content: flex-start; flex-direction: row;"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary museo-slab css-mocked-MuiButtonBase-root-MuiButton-root"
          style="margin: 1rem;"
          tabindex="0"
          type="button"
        >
          Generate Error Summary Email
        </button>
      </div>
    </div>
    <div
      style="display: none;"
    >
      <div
        style="margin: 50px; font-family: Arial Helvetica Sans-Serif; height: 100vh;"
      >
        <h1
          style="text-align: center; margin-bottom: 10px; color: black;"
        >
          RSU Summary
        </h1>
        <br />
        <p
          style="color: black;"
        >
          Below is the generated RSU summary report for all RSUs at 
          2024-10-01T00:00:00.000Z
           UTC:
        </p>
        <div
          style="margin-top: 25px;"
        >
          <div
            class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation2 css-mocked-MuiPaper-root"
            style="position: relative; background-color: white; color: black; font-family: Arial Helvetica Sans-Serif;"
          >
            <div
              class="MuiBox-root css-187halg"
            >
              <div>
                <div
                  style="overflow-y: auto;"
                >
                  <div>
                    <table
                      class="MuiTable-root css-mocked-MuiTable-root"
                      style="table-layout: auto;"
                    >
                      <thead
                        class="MuiTableHead-root css-mocked-MuiTableHead-root"
                      >
                        <tr
                          class="MuiTableRow-root MuiTableRow-head css-mocked-MuiTableRow-root"
                        >
                          <th
                            class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft MuiTableCell-sizeSmall css-mocked-MuiTableCell-root"
                            scope="col"
                            style="box-sizing: border-box;"
                          >
                            <div
                              aria-describedby="rfd-hidden-text-0-hidden-text-0"
                              data-rfd-drag-handle-context-id="0"
                              data-rfd-drag-handle-draggable-id="0"
                              data-rfd-draggable-context-id="0"
                              data-rfd-draggable-id="0"
                              draggable="false"
                              role="button"
                              style="position: relative; min-width: 0; display: flex;"
                              tabindex="0"
                            >
                              <span
                                aria-sort="none"
                                class="MuiButtonBase-root MuiTableSortLabel-root css-mocked-MuiButtonBase-root-MuiTableSortLabel-root"
                                data-testid="mtableheader-sortlabel"
                                role="columnheader"
                                tabindex="0"
                              >
                                RSU
                                <svg
                                  aria-hidden="true"
                                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiTableSortLabel-icon MuiTableSortLabel-iconDirectionAsc css-mocked-MuiSvgIcon-root-MuiTableSortLabel-icon"
                                  data-testid="arrow_downward"
                                  focusable="false"
                                  viewBox="0 0 24 24"
                                >
                                  <path
                                    d="m20 12-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8z"
                                  />
                                </svg>
                              </span>
                              <span
                                data-testid="material-table-column-sort-order-indicator"
                                style="width: 1em;"
                              />
                            </div>
                          </th>
                          <th
                            class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft MuiTableCell-sizeSmall css-mocked-MuiTableCell-root"
                            scope="col"
                            style="box-sizing: border-box;"
                          >
                            <div
                              aria-describedby="rfd-hidden-text-0-hidden-text-0"
                              data-rfd-drag-handle-context-id="0"
                              data-rfd-drag-handle-draggable-id="1"
                              data-rfd-draggable-context-id="0"
                              data-rfd-draggable-id="1"
                              draggable="false"
                              role="button"
                              style="position: relative; min-width: 0; display: flex;"
                              tabindex="0"
                            >
                              <span
                                aria-sort="none"
                                class="MuiButtonBase-root MuiTableSortLabel-root css-mocked-MuiButtonBase-root-MuiTableSortLabel-root"
                                data-testid="mtableheader-sortlabel"
                                role="columnheader"
                                tabindex="0"
                              >
                                Road
                                <svg
                                  aria-hidden="true"
                                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiTableSortLabel-icon MuiTableSortLabel-iconDirectionAsc css-mocked-MuiSvgIcon-root-MuiTableSortLabel-icon"
                                  data-testid="arrow_downward"
                                  focusable="false"
                                  viewBox="0 0 24 24"
                                >
                                  <path
                                    d="m20 12-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8z"
                                  />
                                </svg>
                              </span>
                              <span
                                data-testid="material-table-column-sort-order-indicator"
                                style="width: 1em;"
                              />
                            </div>
                          </th>
                          <th
                            class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft MuiTableCell-sizeSmall css-mocked-MuiTableCell-root"
                            scope="col"
                            style="box-sizing: border-box;"
                          >
                            <div
                              aria-describedby="rfd-hidden-text-0-hidden-text-0"
                              data-rfd-drag-handle-context-id="0"
                              data-rfd-drag-handle-draggable-id="2"
                              data-rfd-draggable-context-id="0"
                              data-rfd-draggable-id="2"
                              draggable="false"
                              role="button"
                              style="position: relative; min-width: 0; display: flex;"
                              tabindex="0"
                            >
                              <span
                                aria-sort="none"
                                class="MuiButtonBase-root MuiTableSortLabel-root css-mocked-MuiButtonBase-root-MuiTableSortLabel-root"
                                data-testid="mtableheader-sortlabel"
                                role="columnheader"
                                tabindex="0"
                              >
                                Online Status
                                <svg
                                  aria-hidden="true"
                                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiTableSortLabel-icon MuiTableSortLabel-iconDirectionAsc css-mocked-MuiSvgIcon-root-MuiTableSortLabel-icon"
                                  data-testid="arrow_downward"
                                  focusable="false"
                                  viewBox="0 0 24 24"
                                >
                                  <path
                                    d="m20 12-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8z"
                                  />
                                </svg>
                              </span>
                              <span
                                data-testid="material-table-column-sort-order-indicator"
                                style="width: 1em;"
                              />
                            </div>
                          </th>
                          <th
                            class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft MuiTableCell-sizeSmall css-mocked-MuiTableCell-root"
                            scope="col"
                            style="box-sizing: border-box;"
                          >
                            <div
                              aria-describedby="rfd-hidden-text-0-hidden-text-0"
                              data-rfd-drag-handle-context-id="0"
                              data-rfd-drag-handle-draggable-id="3"
                              data-rfd-draggable-context-id="0"
                              data-rfd-draggable-id="3"
                              draggable="false"
                              role="button"
                              style="position: relative; min-width: 0; display: flex;"
                              tabindex="0"
                            >
                              <span
                                aria-sort="none"
                                class="MuiButtonBase-root MuiTableSortLabel-root css-mocked-MuiButtonBase-root-MuiTableSortLabel-root"
                                data-testid="mtableheader-sortlabel"
                                role="columnheader"
                                tabindex="0"
                              >
                                SCMS Status
                                <svg
                                  aria-hidden="true"
                                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiTableSortLabel-icon MuiTableSortLabel-iconDirectionAsc css-mocked-MuiSvgIcon-root-MuiTableSortLabel-icon"
                                  data-testid="arrow_downward"
                                  focusable="false"
                                  viewBox="0 0 24 24"
                                >
                                  <path
                                    d="m20 12-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8z"
                                  />
                                </svg>
                              </span>
                              <span
                                data-testid="material-table-column-sort-order-indicator"
                                style="width: 1em;"
                              />
                            </div>
                          </th>
                        </tr>
                      </thead>
                      <tbody
                        class="MuiTableBody-root css-mocked-MuiTableBody-root"
                      >
                        <tr
                          class="MuiTableRow-root css-mocked-MuiTableRow-root"
                          style="height: 49px;"
                        >
                          <td
                            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-mocked-MuiTableCell-root"
                            colspan="4"
                            style="padding-top: 0px; padding-bottom: 0px; text-align: center;"
                          >
                            No records to display
                          </td>
                        </tr>
                      </tbody>
                    </table>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        style="margin: 50px; font-family: Arial Helvetica Sans-Serif; height: 100vh;"
      >
        <h1
          style="text-align: center; margin-bottom: 10px; color: black;"
        >
          RSU Error Summary
        </h1>
        <br />
        <p
          style="color: black;"
        >
          Below is the generated RSU Error summary report for all RSUs at 
          2024-10-01T00:00:00.000Z
           UTC:
        </p>
        <div
          style="margin-top: 25px;"
        >
          <div
            class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation2 css-mocked-MuiPaper-root"
            style="position: relative; background-color: white; color: black; font-family: Arial Helvetica Sans-Serif;"
          >
            <div
              class="MuiBox-root css-187halg"
            >
              <div>
                <div
                  style="overflow-y: auto;"
                >
                  <div>
                    <table
                      class="MuiTable-root css-mocked-MuiTable-root"
                      style="table-layout: auto;"
                    >
                      <thead
                        class="MuiTableHead-root css-mocked-MuiTableHead-root"
                      >
                        <tr
                          class="MuiTableRow-root MuiTableRow-head css-mocked-MuiTableRow-root"
                        >
                          <th
                            class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft MuiTableCell-sizeSmall css-mocked-MuiTableCell-root"
                            scope="col"
                            style="box-sizing: border-box;"
                          >
                            <div
                              aria-describedby="rfd-hidden-text-1-hidden-text-6"
                              data-rfd-drag-handle-context-id="1"
                              data-rfd-drag-handle-draggable-id="0"
                              data-rfd-draggable-context-id="1"
                              data-rfd-draggable-id="0"
                              draggable="false"
                              role="button"
                              style="position: relative; min-width: 0; display: flex;"
                              tabindex="0"
                            >
                              <span
                                aria-sort="none"
                                class="MuiButtonBase-root MuiTableSortLabel-root css-mocked-MuiButtonBase-root-MuiTableSortLabel-root"
                                data-testid="mtableheader-sortlabel"
                                role="columnheader"
                                tabindex="0"
                              >
                                RSU
                                <svg
                                  aria-hidden="true"
                                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiTableSortLabel-icon MuiTableSortLabel-iconDirectionAsc css-mocked-MuiSvgIcon-root-MuiTableSortLabel-icon"
                                  data-testid="arrow_downward"
                                  focusable="false"
                                  viewBox="0 0 24 24"
                                >
                                  <path
                                    d="m20 12-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8z"
                                  />
                                </svg>
                              </span>
                              <span
                                data-testid="material-table-column-sort-order-indicator"
                                style="width: 1em;"
                              />
                            </div>
                          </th>
                          <th
                            class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft MuiTableCell-sizeSmall css-mocked-MuiTableCell-root"
                            scope="col"
                            style="box-sizing: border-box;"
                          >
                            <div
                              aria-describedby="rfd-hidden-text-1-hidden-text-6"
                              data-rfd-drag-handle-context-id="1"
                              data-rfd-drag-handle-draggable-id="1"
                              data-rfd-draggable-context-id="1"
                              data-rfd-draggable-id="1"
                              draggable="false"
                              role="button"
                              style="position: relative; min-width: 0; display: flex;"
                              tabindex="0"
                            >
                              <span
                                aria-sort="none"
                                class="MuiButtonBase-root MuiTableSortLabel-root css-mocked-MuiButtonBase-root-MuiTableSortLabel-root"
                                data-testid="mtableheader-sortlabel"
                                role="columnheader"
                                tabindex="0"
                              >
                                Road
                                <svg
                                  aria-hidden="true"
                                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiTableSortLabel-icon MuiTableSortLabel-iconDirectionAsc css-mocked-MuiSvgIcon-root-MuiTableSortLabel-icon"
                                  data-testid="arrow_downward"
                                  focusable="false"
                                  viewBox="0 0 24 24"
                                >
                                  <path
                                    d="m20 12-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8z"
                                  />
                                </svg>
                              </span>
                              <span
                                data-testid="material-table-column-sort-order-indicator"
                                style="width: 1em;"
                              />
                            </div>
                          </th>
                          <th
                            class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft MuiTableCell-sizeSmall css-mocked-MuiTableCell-root"
                            scope="col"
                            style="box-sizing: border-box;"
                          >
                            <div
                              aria-describedby="rfd-hidden-text-1-hidden-text-6"
                              data-rfd-drag-handle-context-id="1"
                              data-rfd-drag-handle-draggable-id="2"
                              data-rfd-draggable-context-id="1"
                              data-rfd-draggable-id="2"
                              draggable="false"
                              role="button"
                              style="position: relative; min-width: 0; display: flex;"
                              tabindex="0"
                            >
                              <span
                                aria-sort="none"
                                class="MuiButtonBase-root MuiTableSortLabel-root css-mocked-MuiButtonBase-root-MuiTableSortLabel-root"
                                data-testid="mtableheader-sortlabel"
                                role="columnheader"
                                tabindex="0"
                              >
                                Online Status
                                <svg
                                  aria-hidden="true"
                                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiTableSortLabel-icon MuiTableSortLabel-iconDirectionAsc css-mocked-MuiSvgIcon-root-MuiTableSortLabel-icon"
                                  data-testid="arrow_downward"
                                  focusable="false"
                                  viewBox="0 0 24 24"
                                >
                                  <path
                                    d="m20 12-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8z"
                                  />
                                </svg>
                              </span>
                              <span
                                data-testid="material-table-column-sort-order-indicator"
                                style="width: 1em;"
                              />
                            </div>
                          </th>
                          <th
                            class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft MuiTableCell-sizeSmall css-mocked-MuiTableCell-root"
                            scope="col"
                            style="box-sizing: border-box;"
                          >
                            <div
                              aria-describedby="rfd-hidden-text-1-hidden-text-6"
                              data-rfd-drag-handle-context-id="1"
                              data-rfd-drag-handle-draggable-id="3"
                              data-rfd-draggable-context-id="1"
                              data-rfd-draggable-id="3"
                              draggable="false"
                              role="button"
                              style="position: relative; min-width: 0; display: flex;"
                              tabindex="0"
                            >
                              <span
                                aria-sort="none"
                                class="MuiButtonBase-root MuiTableSortLabel-root css-mocked-MuiButtonBase-root-MuiTableSortLabel-root"
                                data-testid="mtableheader-sortlabel"
                                role="columnheader"
                                tabindex="0"
                              >
                                SCMS Status
                                <svg
                                  aria-hidden="true"
                                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiTableSortLabel-icon MuiTableSortLabel-iconDirectionAsc css-mocked-MuiSvgIcon-root-MuiTableSortLabel-icon"
                                  data-testid="arrow_downward"
                                  focusable="false"
                                  viewBox="0 0 24 24"
                                >
                                  <path
                                    d="m20 12-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8z"
                                  />
                                </svg>
                              </span>
                              <span
                                data-testid="material-table-column-sort-order-indicator"
                                style="width: 1em;"
                              />
                            </div>
                          </th>
                        </tr>
                      </thead>
                      <tbody
                        class="MuiTableBody-root css-mocked-MuiTableBody-root"
                      >
                        <tr
                          class="MuiTableRow-root css-mocked-MuiTableRow-root"
                          style="height: 49px;"
                        >
                          <td
                            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-mocked-MuiTableCell-root"
                            colspan="4"
                            style="padding-top: 0px; padding-bottom: 0px; text-align: center;"
                          >
                            No records to display
                          </td>
                        </tr>
                      </tbody>
                    </table>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
